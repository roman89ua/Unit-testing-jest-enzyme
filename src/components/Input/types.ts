export type IInput = {
  onChange: () => void;
  onKeyPress: () => void;
  placeholder?: string;
  value: string;
};
