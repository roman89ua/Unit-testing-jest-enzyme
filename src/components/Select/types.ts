export interface IOption {
  value: number;
  label: string;
}
  
export interface ISelect {
  handleChange: () => void;
  options: IOption[] | null | undefined;
  defaultValue: number | undefined;
}