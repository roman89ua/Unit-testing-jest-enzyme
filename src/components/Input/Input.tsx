import React from 'react';
import { IInput } from './types';

export const Input = ({ onChange, onKeyPress, value, placeholder }: IInput) => {
  return (
    <div className="inputWrapper">
      <i className="fas fa-search" />
      <input
        className="input"
        placeholder={placeholder || ''}
        onChange={onChange}
        onKeyPress={onKeyPress}
        value={value}
      />
    </div>
  );
};
