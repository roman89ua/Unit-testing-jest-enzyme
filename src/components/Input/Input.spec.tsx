import React from 'react';
import { Input } from './Input';
import { shallow, ShallowWrapper } from 'enzyme';
import { IInput } from './types';

describe('Input component', () => {
  const onPress = jest.fn();
  const onChangeInput = jest.fn();

  const mockProps: IInput = {
    onChange: onChangeInput,
    onKeyPress: onPress,
    placeholder: 'Search',
    value: 'value of input',
  };

  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<Input {...mockProps} />);
  });

  it('should render Input component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call onClick method', () => {
    expect(onChangeInput.mock.calls.length).toBe(0);
    component.find('.input').simulate('change');
    expect(onChangeInput.mock.calls.length).toBe(1);
  });

  it('should call onKeyPress method', () => {
    expect(onPress.mock.calls.length).toBe(0);
    component.find('.input').simulate('keypress', { key: 'Enter' }).simulate('keypress', { key: 'Enter' });
    expect(onPress.mock.calls.length).toBe(2);
  });
});
