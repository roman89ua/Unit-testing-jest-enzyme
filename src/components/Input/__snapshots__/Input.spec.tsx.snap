// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render Input component 1`] = `
<div
  className="inputWrapper"
>
  <i
    className="fas fa-search"
  />
  <input
    className="input"
    onChange={[MockFunction]}
    onKeyPress={[MockFunction]}
    placeholder="Search"
    value="value of input"
  />
</div>
`;
